### PR TITLE
feat: support media channels

### PR DESCRIPTION
**Pull request by LordOfPolls**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description

Adds support for media channels - effectively just fancy forum channels

**Please note, media channels are in beta - this pr may need to be updated**

## Changes

- Implement a class for media channels
- Add a mapping for media channels

## Related Issues

https://github.com/discord/discord-api-docs/pull/6232

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [ ] I've updated / added  documentation, where applicable

### CODE REVIEW

**Review by silasary — APPROVED** [review]

File: interactions/models/discord/channel.py [hunk 1]
```diff
@@ -2731,4 +2737,5 @@ def process_permission_overwrites(
     ChannelType.DM: DM,
     ChannelType.GROUP_DM: DMGroup,
     ChannelType.GUILD_FORUM: GuildForum,
+    ChannelType.MEDIA_CHANNEL: GuildMediaChannel,
```
Comment (i0bs) [inline interactions/models/discord/channel.py]: This doesn't seem to follow our current naming conventions, would something like this be better fitting?

```suggestion
    ChannelType.GUILD_MEDIA: GuildMedia,
```

File: interactions/models/discord/enums.py [hunk 2]
```diff
@@ -611,6 +611,8 @@ class ChannelType(CursedIntEnum):
     """Voice channel for hosting events with an audience"""
     GUILD_FORUM = 15
     """A Forum channel"""
+    MEDIA_CHANNEL = 16
```
Comment (i0bs) [inline interactions/models/discord/enums.py]: Addressed likewise with [`channel.py`](https://github.com/interactions-py/interactions.py/pull/1458/files#r1248970595)'s definition:

```suggestion
    GUILD_MEDIA = 16
```

File: interactions/models/discord/enums.py [hunk 3]
```diff
@@ -611,6 +611,8 @@ class ChannelType(CursedIntEnum):
...
+    """Channel that can only contain threads, similar to `GUILD_FORUM` channels"""
```
Comment (i0bs) [inline interactions/models/discord/enums.py:615]: Not one here to judge, but I don't think us comparing it to threads is semantically accurate. The latter seems to be more agreed on, though, what about this instead?

```suggestion
    """Channel that can only contain media files, similar to `GUILD_FORUM` channels"""
```

**Review by i0bs — COMMENTED** [review]
General statements, not necessarily a full review

File: interactions/models/discord/enums.py (hunk 3 above)
Comment (LordOfPolls) [reply interactions/models/discord/enums.py:615]: This is copied verbatim from the dev docs pr

File: interactions/models/discord/channel.py (hunk 1 above)
Comment (LordOfPolls) [reply interactions/models/discord/channel.py]: Sure - cant merge this suggestion as this is a widespread change

**Review by i0bs — APPROVED** [review]

**Pull request merged**